cadc-dali/src/main/java/ca/nrc/cadc/dali/tables/parquet/ParquetReader.java

```diff
@@ -0,0 +1,190 @@
+package ca.nrc.cadc.dali.tables.parquet;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.avro.generic.GenericRecord;
+
+import org.apache.log4j.Logger;
+
+import org.apache.parquet.avro.AvroParquetReader;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.io.InputFile;
+import org.apache.parquet.io.SeekableInputStream;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.Type;
+
+
+//TODO: Temporary implementation
+public class ParquetReader {
+
+    private static final Logger log = Logger.getLogger(ParquetReader.class);
+
+    public TableShape read(InputStream inputStream) throws IOException {
+
+        InputFile inputFile = getInputFile(inputStream);
+        int recordCount = 0;
+        int columnCount = 0;
+
+        ParquetMetadata metadata = ParquetFileReader.open(inputFile).getFooter();
+        MessageType schema = metadata.getFileMetaData().getSchema();
+        columnCount = schema.getFieldCount();
+
+        String votable = metadata.getFileMetaData().getKeyValueMetaData().get("votable");
+        log.debug("VOTable: " + votable);
+
+        try (org.apache.parquet.hadoop.ParquetReader<GenericRecord> reader = AvroParquetReader.<GenericRecord>builder(inputFile).build()) {
+            GenericRecord record;
+
+            while ((record = reader.read()) != null) {
+                log.debug("Reading Record: " + record);
+                recordCount++;
+
+                // TODO: This is a temporary implementation to read the data
+                Map<String, Object> rowData = new HashMap<>();
+                for (Type field : schema.getFields()) {
+                    String fieldName = field.getName();
+                    rowData.put(fieldName, record.get(fieldName));
+                }
+                log.debug("Retrieved Row Data: " + rowData);
+            }
+            log.debug("Record Count = " + recordCount);
+
+        } catch (Exception e) {
+            throw new IOException("failed to read parquet data", e);
+        }
+        if (recordCount == 0) {
+            throw new RuntimeException("NO Records Read");
+        }
+        return new TableShape(recordCount, columnCount);
+    }
+
+    private static InputFile getInputFile(InputStream inputStream) throws IOException {
+
+        // Read the entire InputStream into a byte array
```

Comment:
OK for now, but we can't use this class with any real (large) content without considerable thought.